dashboard/src/actions/overviewActions.js

```diff
@@ -129,14 +129,14 @@ export const updateDataset =
           (item) => item.resource_id === dataset.resource_id
         );
         runs[dataIndex].metadata[metaDataActions[actionType]] =
-          response.data[metaDataActions[actionType]];
+          response.data.metadata[metaDataActions[actionType]];
         dispatch({
           type: TYPES.USER_RUNS,
           payload: runs,
         });
         dispatch(initializeRuns());
       } else {
-        dispatch(showToast(DANGER, response?.data?.message));
+        dispatch(showToast(DANGER, response?.data?.errors));
```

Comment:
There are two small points to consider here:
1. Partial success, which is unlikely but possible: you'll get a 200 response, but there might still be a non-empty `errors` list in the JSON.
2. On failure, you'll get the 400 response, a summary `message` ("Unable to modify any specified metadata keys") *and* the errors for each key. What does `showToast` do with a JSON object? It'll be something like `{"user.dashboard.favorite": "Something went wrong", "global.dashboard.seen": "something else went wrong"}` ... how will this be displayed?

More importantly, I'm worried here about the use of conditional chaining. On a failure here the server should always return an `errors` field, but if the call fails before it hits the server, with something like a connection error, there may be no `response.data.errors` or even `response.data`; presumably this is why you're using conditional chaining in the first place, but the real problem is that, if this happens, we end up with `showToast(DANGER, undefined)` ... which isn't very useful. Maybe the `showToast` itself could recognize a falsey value and subsitute some generic message, but it doesn't have context to be very smart about it. In any case, we really don't want to pop up blank toasts!

This probably requires a general fix outside the scope of this PR (perhaps passing `showToast` the response object, for example?), but that doesn't mean you can't do something a little "safer" while you're changing this call. Even if maybe it's just a fallback "Problem updating metadata" if not `response?.data?.errors`...

Reply:
Addressed the mentioned points